akka-javasdk/src/main/java/akka/javasdk/agent/SessionMemoryEntity.java

```diff
@@ -268,27 +279,59 @@ public Effect<Done> compactHistory(CompactionCmd cmd) {
               toolCallResponse.componentId(),
               toolCallResponse.id(),
               toolCallResponse.name(),
-              toolCallResponse.text()));
+              toolCallResponse.text(),
+                toolCallResponse.size()));
           }
 
           case AiMessage aiMessage -> {
             events.add(new Event.AiMessageAdded(
               aiMessage.timestamp(),
               aiMessage.componentId(),
               aiMessage.text(),
-              aiMessage.inputTokens(),
-              aiMessage.outputTokens(),
+                aiMessage.size(),
+                0L, // filled in later
               aiMessage.toolCallRequests()));
           }
         }
       });
     }
 
+    var eventsWithSize = updateHistorySize(events);
+
     return effects()
-        .persistAll(events)
+        .persistAll(eventsWithSize)
         .thenReply(__ -> Done.done());
   }
 
+  private List<Event> updateHistorySize(List<Event> events) {
+    var result = new ArrayList<Event>();
+    var size = currentState().currentSizeInBytes;
```

Comment:
Is this used? I mean the `currentState().currentSizeInBytes` . I thought since the first msg in event is always `HistoryCleared` this goes straight to 0L always.. right? So it can be declared as 0L?